File: quixo/new_server/models/Product.model.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    name: { type: String, required: true },
    brand: { type: String },
    shortDescriptions: { type: String },
    description: { type: String },
    pricePerUnit: { type: Number, required: true },
    unit: { type: String }, // kg, pc, etc.
    discount: { type: Number, default: 0 },
    productCategory: { type: mongoose.Schema.Types.ObjectId, ref: 'Category' },
    deliveryCategory: { type: String, enum: ['quick', 'ecommerce'] },
    stock: { type: Number, default: 0 },
    photos: [{ type: String }],
    vendorId: { type: mongoose.Schema.Types.ObjectId, ref: 'Vendor', required: true },
    rating: { type: Number, default: 0 },
    approved: { type: Boolean, default: false },
    hidden: { type: Boolean, default: false }
}, { timestamps: true });

module.exports = mongoose.model('Product', productSchema);

File: quixo/flutter/mart_vendorapp/lib/features/Product/data/product_model.dart
class Product {
  final String id;
  final String name;
  final String shortDescription;
  final double pricePerUnit;
  final String productCategory;
  final String deliveryCategory;
  final int stock;
  final String brandName;
  final List<String> photos;

  Product({
    required this.id,
    required this.name,
    required this.shortDescription,
    required this.pricePerUnit,
    required this.productCategory,
    required this.deliveryCategory,
    required this.stock,
    required this.brandName,
    this.photos = const [],
  });

  factory Product.fromMap(Map<String, dynamic> map) {
    List<String> photoList = [];
    final ph = map["photos"];
    if (ph is List) {
      photoList = ph.map((e) => e.toString()).toList();
    }
    return Product(
      id: (map["Product id"] ?? map["product id"] ?? '').toString(),
      name: map["name"] ?? "",
      shortDescription: map["short description"] ?? "",
      pricePerUnit: (map["price per unit"] ?? 0).toDouble(),
      productCategory:
          map["product catagory"]?.toString() ?? map["product category"]?.toString() ?? "",
      deliveryCategory: map["delivary category"]?.toString() ?? "",
      stock: (map["stock"] is int) ? map["stock"] as int : int.tryParse('${map["stock"] ?? 0}') ?? 0,
      brandName: map["brand name"]?.toString() ?? "",
      photos: photoList,
    );
  }
}